src/main/java/org/onedatashare/server/module/box/BoxResource.java

```diff
@@ -0,0 +1,379 @@
+                stat.dir = false;
+                stat.size = info.getSize();
+            }
+            stat.id = info.getID();
+            stat.name = info.getName();
+            if (info.getContentModifiedAt() != null)
+                stat.time = info.getContentModifiedAt().getTime() / 1000;
+        } catch (BoxAPIException e) {
+            e.printStackTrace();
+        }
+        return stat;
+    }
+
+    public BoxTap tap() {
+        BoxTap boxTap = new BoxTap();
+        return boxTap;
+    }
+
+    class BoxTap implements Tap {
+        long size;
+        BoxAPIConnection api = session.client;
+        BoxAPIRequest req;
+
+        @Override
+        public Flux<Slice> tap(Stat stat, long sliceSize) {
+            BoxFile file = new BoxFile(api, stat.id);
+            URL downloadUrl = file.getDownloadURL();
+            try {
+                req = new BoxAPIRequest(api, downloadUrl, HttpMethod.GET);
+            }catch(Exception e){
+                e.printStackTrace();
+            }
+            size = stat.getSize();
+            return tap(sliceSize);
+
+        }
+
+        @Override
+        public Flux<Slice> tap(long sliceSize) {
+
+            int sliceSizeInt = Math.toIntExact(sliceSize);
+            int sizeInt = Math.toIntExact(size);
+            BoxAPIResponse resp = req.send();
+            InputStream inputStream = resp.getBody();
+            return Flux.generate(
+                    () -> 0,
+                    (state, sink) -> {
+                        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                        if (state + sliceSizeInt < sizeInt) {
+                            try {
+                                IOUtils.copy(inputStream, outputStream);
+                            } catch (IOException e) {
+                                e.printStackTrace();
+                            }
+                            sink.next(new Slice(outputStream.toByteArray()));
+                            try {
+                                outputStream.close();
+                            } catch (IOException e) {
+                                e.printStackTrace();
+                            }
+                        } else {
+                            try {
+                                IOUtils.copy(inputStream, outputStream);
+                            } catch (IOException e) {
+                                e.printStackTrace();
+                            }
+
+                            sink.next(new Slice(outputStream.toByteArray()));
+                            sink.complete();
+                        }
+                        return state + sliceSizeInt;
+                    });
+
+        }
+
+
+    }
+    public BoxDrain sink() {
+        return new BoxDrain().start();
+    }
+
+    public BoxDrain sink(Stat stat){
+        return new BoxDrain().start(path + stat.getName());
+    }
+
+    class BoxDrain implements Drain {
```

Comment:
Assuming Drain will be done in the next PR.

Reply:
Yup, Drain is coming up next once I get everything updated